src/loaders/index.ts

```diff
@@ -2,20 +2,26 @@ import { IVolumeLoader } from "./IVolumeLoader";
 
 import { OMEZarrLoader } from "./OmeZarrLoader";
 import { JsonImageInfoLoader } from "./JsonImageInfoLoader";
+import { RawArrayLoader, RawArrayData, RawArrayInfo } from "./RawArrayLoader";
 import { TiffLoader } from "./TiffLoader";
 import VolumeCache from "../VolumeCache";
 
 export const enum VolumeFileFormat {
   ZARR = "zarr",
   JSON = "json",
   TIFF = "tiff",
+  DATA = "data",
 }
 
+// superset of all necessary loader options
 export type CreateLoaderOptions = {
   fileType?: VolumeFileFormat;
   cache?: VolumeCache;
   scene?: number;
   concurrencyLimit?: number;
+  // optional raw image data for VolumeFileFormat.DATA
+  imageData?: RawArrayData;
+  imageDataInfo?: RawArrayInfo;
```

Comment:
I know it could be overkill, but it just looks weird to have optional things that then throw an error if they don't exist. but I get that they're only required in that one case

Reply:
I wrote the first version of this file and did initially try to write this type as a union of option objects for each `fileType`, but gave up (motivated in part by some consulation with Dan) after realizing that all the options were optional anyways. But I definitely agree that this type shouldn't allow you to produce errors by not providing these options when they're needed - perhaps by unioning one options object for `VolumeFileFormat.DATA` and one for `Exclude<VolumeFileFormat, VolumeFileFormat.DATA>`.

In fact, should `createVolumeLoader` let you create a data loader in the first place? Basically the primary convenience it provides is accepting a URL to a data source of undetermined format and working out which kind of loader to hand back. I can't think of a likely scenario where a client might not know whether a volume is coming out of a local array or a remote source, and having to pass in an unused `path` argument is a bit clumsy anyways.

Reply:
Do you want a `CreateLoaderOptionsFile` and `CreateLoaderOptionsRawArray`?  I could try to do that.  What would satisfy this PR comment?